src/monodroid/jni/embedded-assemblies.cc

```diff
@@ -215,50 +227,48 @@ EmbeddedAssemblies::open_from_bundles (MonoAssemblyName* aname, bool ref_only)
 		log_info_nocheck (LOG_ASSEMBLY, "open_from_bundles: loaded assembly: %p\n", a);
 	}
 
-#if defined (NET6) && defined (NET6_ALC_WORKS)
-	if (error != nullptr) {
-		error->error_code = a == nullptr ? MONO_ERROR_NONE : MONO_ERROR_FILE_NOT_FOUND;
-	}
-#endif // def NET6 && def NET6_ALC_WORKS
 	return a;
 }
 
-#if defined (NET6) && defined (NET6_ALC_WORKS)
+#if defined (NET6)
 MonoAssembly*
 EmbeddedAssemblies::open_from_bundles (MonoAssemblyLoadContextGCHandle alc_gchandle, MonoAssemblyName *aname, [[maybe_unused]] char **assemblies_path, [[maybe_unused]] void *user_data, MonoError *error)
 {
-	log_warn (LOG_DEFAULT, __PRETTY_FUNCTION__);
 	return embeddedAssemblies.open_from_bundles (aname, alc_gchandle, error);
 }
-#else // def NET6 && def NET6_ALC_WORKS
+#else // def NET6
 MonoAssembly*
-EmbeddedAssemblies::open_from_bundles_full (MonoAssemblyName *aname, [[maybe_unused]] char **assemblies_path, [[maybe_unused]] void *user_data)
+EmbeddedAssemblies::open_from_bundles_refonly (MonoAssemblyName *aname, [[maybe_unused]] char **assemblies_path, [[maybe_unused]] void *user_data)
 {
-	return embeddedAssemblies.open_from_bundles (aname, false);
+	return embeddedAssemblies.open_from_bundles (aname, true);
 }
+#endif // ndef NET6
 
 MonoAssembly*
-EmbeddedAssemblies::open_from_bundles_refonly (MonoAssemblyName *aname, [[maybe_unused]] char **assemblies_path, [[maybe_unused]] void *user_data)
+EmbeddedAssemblies::open_from_bundles_full (MonoAssemblyName *aname, [[maybe_unused]] char **assemblies_path, [[maybe_unused]] void *user_data)
 {
-	return embeddedAssemblies.open_from_bundles (aname, true);
+	return embeddedAssemblies.open_from_bundles (aname, false);
 }
-#endif // !(def NET6 && def NET6_ALC_WORKS)
 
 void
-EmbeddedAssemblies::install_preload_hooks ()
+EmbeddedAssemblies::install_preload_hooks (bool early)
```

Comment:
This somewhat "irks" me in that "early" doesn't mean anything.  The fact that it's `bool` implies that `install_preload_hooks()` is called twice -- which it is -- but *why* it's called twice, or how the two call-sites differ, isn't fully implied.  (One is "early", presumably; but *what does that **mean***?)

Additionally, given that I can't see *any* code share between the `early` and `!early` blocks, I think it would be more understandable if we split this up into two separately named methods, which would allow us to use a more "meaningful" adjective than "early".

Reply:
I will split the method into two separate ones.